include/flang/Semantics/tools.h

```diff
@@ -49,6 +49,7 @@ const DeclTypeSpec *FindParentTypeSpec(const DerivedTypeSpec &);
 const DeclTypeSpec *FindParentTypeSpec(const DeclTypeSpec &);
 const DeclTypeSpec *FindParentTypeSpec(const Scope &);
 const DeclTypeSpec *FindParentTypeSpec(const Symbol &);
+bool IsSequenceType(const DeclTypeSpec *type);
```

Comment:
You might want to make the argument a reference rather than a pointer, unless you expect that this new predicate would be meaningfully called with a null pointer.

Reply:
Thanks, Peter.  I plan to follow Tim's suggestion to make this a member function of `DeclTypeSpec`.